### PR TITLE
Integrates SQLAlchemy into database_wrapper & Implements AudioSampleMetaData REST endpoint

### LINKED ISSUE #38: implement the logic to store the wakeword audio file metadata into the database
## Objective
Support progress toward the [Implement Audio-File-Upload....][1] by _saving the metadata of the files_

## Key Result
Commit code that saves the wakeword audio file metadata into the database

## Additional Details
```
    metadata = {
        "isWakeWord": True,
        "firstName": "john",
        "lastName": "doe",
        "gender": "m",
        "noiseLevel": "q",
        "location": "here",
        "tone": "serious-but-not-really",
        "timestamp": 1577077883,
        "username": "guest"
        "filename": "ww_q_serious-but-not-really_here_m_doe_john_1577077883_guest.wav"
    }
```

[1]: https://github.com/calpoly-csai/api/milestone/1

### PR DESCRIPTION
**Pull request by mfekadu**
### Summary
* SQLAlchemy needs the port (3306) so that is now added to config.json
* the `sqlalchemy_demo.py` serves as a quickstart with quicklinks to documentation
* modified Courses to be structured the SQLAlchemy way
* added an __init__.py to make Entity a module
* created NumbusMySQSLAlchemy as an interface for our database that is backed by SQLAlchemy and hopefully this "object" can serve as a data access object (DAO)

### A bunch of code happened thanks to [~pair~](https://en.m.wikipedia.org/wiki/Pair_programming) [mob programming](https://en.m.wikipedia.org/wiki/Mob_programming)
Thank you friends!
* `AudioSampleMetaData` is a new class that represents the corresponding table in the database
* `create_AudioSampleMetaData_table` is a new function that creates that table by using SQLAlchemy to automatically generate the right SQL CREATE statement
* `NoiseLevel` is a new Enum that represents quiet | medium | loud
* `save_audio_sample_meta_data` is a new function that uses SQLAlchemy to magically insert the new metadata object given in JSON format

### Extras
* filenames are also stored into the database
* there's a bunch of fancy error handling

### `@app.route('/new_data/wakeword', methods=['POST'])`
* that's the endpoint

### New Dependencies
[`SQLAlchemy`](https://www.sqlalchemy.org) because it's a very good package for [ORM](https://en.wikipedia.org/wiki/Object-relational_mapping). In other words, it makes working with the Nimbus database a lot easier. 

## Type of change (pick-one)
- [x] New feature (_non-breaking change which adds functionality_)
- [x] This change requires a documentation update

## How Has This Been Tested?
See the gifs below

## Checklist (check-all-before-merge)
_formatting help: `- [x]` means "checked' and `- [ ]` means "unchecked"_

- [ ] I documented my code according to the [Google Python Style Guide][1]

- [ ] I ran `./build_docs.sh` and the docs look fine **cannot because not `build_docs` not implemented**

- [ ] I ran `./type_check.sh` and got no errors **cannot because not `type_check` not implemented**

- [x] I ran `./format.sh` because it automatically cleans my code for me 😄 

- [x] I ran `./lint.sh` to check for what "format" missed

- [ ] I added my tests to the `/tests` directory **TODO**

- [ ] I ran `./run_tests.sh` and all the tests pass  **TODO**

[1]: https://google.github.io/styleguide/pyguide.html

### CODE REVIEW

**Comment by mfekadu** [comment]
# This is how I tested it...

## Precondition - I commented out the `save_audiofile` function because it was not a subject under test
```
# save_audiofile(filename, request.files["wav_file"])
```

Feb-02-2020 22-09-20

**Comment by mfekadu** [comment]
# another manual testing.gif ...

Feb-02-2020 22-10-23

**Review by Waidhoferj — APPROVED** [review]
@mfekadu Amazing documentation and pull request description. The most friendly ux ever for devs I've ever seen in this context 👍. Code looks fine, but your linter is failing. I'll approve but probably should take a look at the error its throwing.

**Comment by mfekadu** [comment]
@Waidhoferj Thank you for your kind words! 

Yes, I agree there are issues with style check that I need to resolve in a separate PR since this one is large enough already.

File: Entity/OfficeHours.py [hunk 1]
```diff
@@ -1,10 +1,7 @@
...
 
-   def __init__(self, Professors_id, ohroom, ohday, ohtime):
-      self.Professors_id = Professors_id
-      self.ohroom = ohroom
-      self.ohday = ohday
-      self.ohtime = ohtime
-      
+    def __init__(self, Professors_id, ohroom, ohday, ohtime):
+        self.Professors_id = Professors_id
+        self.ohroom = ohroom
```
Comment (snekiam) [inline Entity/OfficeHours.py:5]: Can we add a comment for what these variables are? I had trouble figuring out from the name (despite the class being named 'OfficeHours'

**Review by snekiam — COMMENTED** [review]
Everything looks really good! Just one small comment.  I don't think I've ever seen a pull request with a gif of someone testing it, great job everyone who worked hard on this!

File: Entity/OfficeHours.py (hunk 1 above)
Comment (mfekadu) [reply Entity/OfficeHours.py:5]: @snekiam, I agree! Please do feel empowered to make that change!

**Pull request merged**